src/status_im2/contexts/chat/photo_selector/album_selector/view.cljs

```diff
@@ -0,0 +1,58 @@
+(ns status-im2.contexts.chat.photo-selector.album-selector.view
+  (:require
+   [quo2.core :as quo]
+   [react-native.core :as rn]
+   [utils.i18n :as i18n]
+   [utils.re-frame :as rf]
+   [quo2.foundations.colors :as colors]
+   [status-im2.contexts.chat.photo-selector.view :refer [album-title]]
+   [status-im2.contexts.chat.photo-selector.album-selector.style :as style]))
+
+(defn album
+  [{:keys [title count uri]} _ _ selected-album]
+  (let [selected? (= selected-album title)]
+    [rn/touchable-opacity
+     {:on-press (fn []
+                  (rf/dispatch [:chat.ui/camera-roll-select-album title])
+                  (rf/dispatch [:navigate-back]))
+      :style    (style/album-container selected?)}
+     [rn/image
+      {:source {:uri uri}
+       :style  style/cover}]
+     [rn/view {:style {:margin-left 12}}
+      [quo/text {:weight :medium} title]
+      [quo/text
+       {:size  :paragraph-2
+        :style {:color (colors/theme-colors colors/neutral-50 colors/neutral-40)}}
+       (str count " " (i18n/label :t/images))]]
+     (when selected?
+       [rn/view
+        {:style {:position :absolute
+                 :right    16}}
+        [quo/icon :i/check
+         {:size 20 :color (colors/theme-colors colors/primary-50 colors/primary-60)}]])]))
+
+(defn section-header
+  [{:keys [title]}]
+  (when (not= title :smart-albums)
+    [quo/divider-label
+     {:label           title
+      :container-style style/divider}]))
+
+(defn album-selector
+  []
+  (rf/dispatch [:chat.ui/camera-roll-get-albums])
```

Comment:
this is not a proper way, did-mount or effect should be used

Reply:
but im not sure cc @ilmotta wdyt ?

Reply:
I remember the original PR that @OmarBasem opened where I asked the same thing. Nowadays I'm starting to think that this is solution to dispatch on mount is quite acceptable (?)

Reasoning: if we accept that declaring a local ratom in a form-2 component is good (we know it is, it's in the Reagent docs too), then why wouldn't dispatching events be acceptable? The dispatch is guaranteed to run on mount, but without the verbosity of the `reagent.core/create-class`.

I couldn't find documentation about this solution in re-frame, would love to find something.

Reply:
Yes this looks odd I did not notice it. Moved it to a hook. I am wondering how did it work tho? What does that part after `defn` and before `fn` represent in react, is it the component constructor?

Reply:
It's in the Reagent docs @OmarBasem. It's the part that explains how form-2 components work. But how this all works is relatively complicated.

Here's one important part of the Reagent implementation:

https://github.com/reagent-project/reagent/blob/6d2b0d2a1dbbdaca96840e2e09c5d50edcdced94/src/reagent/impl/component.cljs#L396

You can see Reagent has to store the anonymous function (the so called renderer) and reuse it across future renders. The parent function of `defn` runs only once per mount. As the source code shows, Reagent uses certain React features, like `useRef`, but the logic to use renderers and not rely on React lifecycles is entirely managed by Reagent (not React).

This is not a hack, it's actually very convenient for Reagent developers that they don't need to use `hooks` and `:f>` just to do something on mount (like a dispatch or creating a ratom).

It's fine you used a hook, but in Reagent we don't have to for the *on mount* lifecycle. Thanks to you I learned a bit more about the internals of Reagent, thanks!

Reply:
Thanks @ilmotta for the valuable information!